Robot-Tests/KW_Leveys.robot

```diff
@@ -0,0 +1,251 @@
+    Element Should Be Visible                  ${Leveys_leikkausnäkymä_B}
+
+    Select Radio Button                         road-width-a    disabled
+    Radio Button Should Be Set To               road-width-a    disabled
+    Select Radio Button                         road-width-b    disabled
+    Radio Button Should Be Set To               road-width-b    disabled
+
+    Element Should Be Enabled                   ${FA_footer_Tallenna}
+    Click Element                               ${FA_footer_Tallenna}
+
+    Select Radio Button                         road-width-a    enabled
+    Radio Button Should Be Set To               road-width-a    enabled
+    Select Radio Button                         road-width-b    enabled
+    Radio Button Should Be Set To               road-width-b    enabled
+    
+    Element Should Be Enabled                   ${FA_footer_Peruuta}
+    Click Element                               ${FA_footer_Peruuta}
+
+
+Leveystieto usealle linkille    [arguments]    ${testipaikka}
+    Log    Usean leveystiedon lisäämistapojen testaaminen
+    Vaihda Tietolaji                            ${TL_Leveys_RB}
+    Paikanna osoite                             ${testipaikka}
+    #Zoomaa kartta                               1  200 m
+
+    Odota sivun latautuminen
+    Siirry Muokkaustilaan
+
+    #Leveyden lisääminen laatikkotyökalulla
+    Wait Until Element Is Enabled               ${Muokkaustila_Laatikko}
+    Click Element                               ${Muokkaustila_Laatikko}
+    Suorita laatikkovalinta
+    Wait Until Element Is Visible               road-width
+    Select Radio Button                         road-width    enabled
+    Radio Button Should Be Set To               road-width    enabled
+    Wait Until Element Is Visible               ${Leveys_tekstikenttä}
+    Input Text                                  ${Leveys_tekstikenttä}    400    True
+    Click Button                                ${FA_header_Tallenna}
+
+    #Muutoksen peruutus (laatikkotyökalu)
+    Element Should Be Visible                   ${Muokkaustila_Laatikko}
+    Click Element                               ${Muokkaustila_Laatikko}
+    Suorita laatikkovalinta
+    Wait Until Element Is Visible               road-width
+    Click Button                                ${FA_header_Tallenna}
+
+    #Leveyden lisääminen polygontyökalulla
+    Wait Until Element Is Enabled               ${Muokkaustila_PolygonTool}
+    Click Element                               ${Muokkaustila_PolygonTool}
+    Suorita monivalinta
+    Wait Until Element Is Visible               road-width
+    Select Radio Button                         road-width    enabled
+    Radio Button Should Be Set To               road-width    enabled
+    Wait Until Element Is Visible               ${Leveys_tekstikenttä}
+    Input Text                                  ${Leveys_tekstikenttä}    400    True
+    Click Button                                ${FA_header_Tallenna}
+
```

Comment:
Ja täällä myös tarkistus, että lisätty leveys löytyy.

Reply:
Lisätty

Reply:
Jos halutaan tsekata, että tallennus toimii, niin mun mielestä kannattaisi tallennuksen jälkeen valita linkki uudestaan, ja siirtää toi should sen jälkeen. Nyt taitaa testata vain että input on mitä pitää.

Vaihtoehtoisesti: Jos koet tämän tarpeettomana, niin tallennuksen voi ottaa tästä kohtaa kokonaan pois, niin saadaan testi ajettua nopeammin.

Reply:
Sama kuin edelliseen